sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3254,12 +3256,14 @@ pi_result piMemRelease(pi_mem Mem) {
 
   if (Mem->isImage()) {
     char *ZeHandleImage;
-    PI_CALL(Mem->getZeHandle(ZeHandleImage, _pi_mem::write_only));
-    auto ZeResult = ZE_CALL_NOCHECK(
-        zeImageDestroy, (pi_cast<ze_image_handle_t>(ZeHandleImage)));
-    // Gracefully handle the case that L0 was already unloaded.
-    if (ZeResult && ZeResult != ZE_RESULT_ERROR_UNINITIALIZED)
-      return mapError(ZeResult);
+    auto Image = static_cast<pi_image>(Mem);
+    if (Image->OwnZeMemHandle) {
+      PI_CALL(Mem->getZeHandle(ZeHandleImage, _pi_mem::write_only));
```

Comment:
NOTE: there is no support for _pi_image in contexts with multiple devices (only for _pi_buffer)
So the image can only be used on the devices where it was created and not on any other device in the same context.
Perhaps, we need to document this limitation somehow.

Reply:
Greg brought that up some time ago and we mentioned it in the docs .md for make_image.

Reply:
I see, good, left a comment there.